examples/quantization_w8a8_fp8/README.md

```diff
@@ -22,122 +20,102 @@ The example includes an end-to-end script for applying the quantization algorith
-MAX_SEQUENCE_LENGTH=2048
-
-# Load dataset.
-ds = load_dataset("HuggingFaceH4/ultrachat_200k", split="train_sft")
-ds = ds.shuffle(seed=42).select(range(NUM_CALIBRATION_SAMPLES))
+### 2) Apply Quantization
 
-# Preprocess the data into the format the model is trained with.
-def preprocess(example):
-    return {"text": tokenizer.apply_chat_template(example["messages"], tokenize=False,)}
-ds = ds.map(preprocess)
+For `fp8` quantization, we can recover accuracy with simple PTQ quantization.
 
-# Tokenize the data (be careful with bos tokens - we need add_special_tokens=False since the chat_template already added it).
-def tokenize(sample):
-    return tokenizer(sample["text"], padding=False, max_length=MAX_SEQUENCE_LENGTH, truncation=True, add_special_tokens=False)
-ds = ds.map(tokenize, remove_columns=ds.column_names)
-```
-
-### 3) Apply Quantization
+We recommend targeting all `Linear` layers using the `FP8_Dynamic` scheme, which uses:
+- Static, per-channel quantization on the weights
+- Dynamic, per-token quantization on the activations
 
-With the dataset ready, we will now apply quantization.
-
-We first select the quantization algorithm. In our case, we will apply the default recipe for `fp8` (which uses static-per-tensor weights and static-per-tensor activations) to all linear layers.
-> See the `Recipes` documentation for more information on making complex recipes
+Since simple PTQ does not require data for weight quantization and the activations are quantized dynamically, we do not need any calibration data for this quantization flow.
 
 ```python
 from llmcompressor.transformers import oneshot
 from llmcompressor.modifiers.quantization import QuantizationModifier
 
 # Configure the quantization algorithm to run.
-recipe = QuantizationModifier(targets="Linear", scheme="FP8", ignore=["lm_head"])
+recipe = QuantizationModifier(targets="Linear", 
+                              scheme="FP8_Dynamic",
+                              ignore=["lm_head"])
 
 # Apply quantization.
-oneshot(
-    model=model,
-    dataset=ds,
-    recipe=recipe,
-    max_seq_length=MAX_SEQUENCE_LENGTH,
-    num_calibration_samples=NUM_CALIBRATION_SAMPLES,
-)
+oneshot(model=model, recipe=recipe)
 
 # Save to disk compressed.
-SAVE_DIR = MODEL_ID.split("/")[1] + "-W8A8-FP8"
-model.save_pretrained(SAVE_DIR, save_compressed=True)
+SAVE_DIR = MODEL_ID.split("/")[1] + "-FP8-Dynamic"
+model.save_pretrained(SAVE_DIR)
 tokenizer.save_pretrained(SAVE_DIR)
 ```
 
 We have successfully created an `fp8` model!
 
-### 4) Evaluate Accuracy
+### 3) Evaluate Accuracy
+
+#### Install
 
-With the model created, we can now load and run in vLLM (after installing).
+Neural Magic's fork of `lm-evaluation-harness` implements the evaluation strategy used by Meta in the Llama3.1 launch. You can install this branch from source below:
+
+```bash
+pip install vllm
+pip install git+https://github.com/neuralmagic/lm-evaluation-harness.git@a0e54e5f1a0a52abaedced474854ae2ce4e68ded
+```
```

Comment:
It may be best to use a task that doesn't require a fork of lm-eval to reproduce results. AFAIK it is only ARC-C and GSM8k that require these custom changes. Winogrande is pretty fast, so maybe use that with lm-eval==0.4.3

Reply:
I like GSM because its easy to understand + its a good proof point for users that its working in a generative task

Reply:
Do you need to have the reproduction of paper results then? I think we shouldn't push people towards our fork if possible. I think it also makes the most realistic example to show evals for both the unquantized and quantized checkpoint, so it shouldn't matter to get this specific COT setup.